app/src/main/java/com/xpay/kotlinutils/XpayUtils.kt

```diff
@@ -92,30 +106,36 @@ object XpayUtils {
         })
     }
 
+   
     fun pay(
-        amount: Number,
-        token: String,
         onSuccess: (PayResponse) -> Unit,
         onFail: (String) -> Unit
     ) {
         val user: User = user!!
-        val billingData: HashMap<String, Any> = HashMap<String, Any>()
-        val requestBody: HashMap<String, Any> = HashMap<String, Any>()
-
+        val billingData: HashMap<String, Any> = HashMap()
+        val requestBody: HashMap<String, Any> = HashMap()
+        
+        when (payUsing) {
+            "CARD" -> totalAmount?.card
+            "CASH" -> totalAmount?.cash
+            "KIOSK" -> totalAmount?.kiosk
+        }
         billingData["name"] = user.name
         billingData["email"] = user.email
         billingData["phone_number"] = user.phone
-        requestBody["amount"] = amount
+        requestBody["amount"] = amount!!
         currency?.let { requestBody.put("currency", it) }
         variableAmountID?.let { requestBody.put("variable_amount_id", it) }
         communityId?.let { requestBody.put("community_id", it) }
-        payUsing?.let { requestBody.put("pay_using", it) }
+        payUsing.let {
+            if (it != null) {
+                requestBody["pay_using"] = it
+            }
+        }
         requestBody["billing_data"] = billingData
-        requestBody["custom_fields"] = customFields
```

Comment:
check if the list length is bigger than zero and put custom field list in payload

Reply:
this